psi4/src/psi4/libmints/oeprop.cc

```diff
@@ -1784,6 +1791,597 @@ PopulationAnalysisCalc::compute_lowdin_charges(bool print_output) {
+    };
+    params[28] = {
+        std::make_tuple(1.01282, 80.98814),
+        std::make_tuple(6.43837, 17.26336),
+        std::make_tuple(16.92012, 4.03115),
+        std::make_tuple(3.62869, 1.36133)
+    };
+    params[29] = {
+        std::make_tuple(1.01839, 83.81831),
+        std::make_tuple(6.47823, 17.85149),
+        std::make_tuple(18.65720, 4.05312),
+        std::make_tuple(2.84618, 1.37570)
+    };
+    params[30] = {
+        std::make_tuple(1.00931, 87.16777),
+        std::make_tuple(6.27682, 18.84319),
+        std::make_tuple(18.99747, 4.35989),
+        std::make_tuple(3.71640, 1.44857)
+    };
+    params[31] = {
+        std::make_tuple(1.00600, 90.34057),
+        std::make_tuple(6.16315, 19.71091),
+        std::make_tuple(19.81836, 4.57852),
+        std::make_tuple(4.01249, 1.29122)
+    };
+    params[32] = {
+        std::make_tuple(0.99467, 93.80965),
+        std::make_tuple(5.91408, 20.85993),
+        std::make_tuple(19.89501, 4.95158),
+        std::make_tuple(5.19624, 1.39361)
+    };
+    params[33] = {
+        std::make_tuple(0.98548, 97.22822),
+        std::make_tuple(5.68319, 22.01684),
+        std::make_tuple(19.83497, 5.33969),
+        std::make_tuple(6.49637, 1.51963)
+    };
+    params[34] = {
+        std::make_tuple(0.97822, 100.60094),
+        std::make_tuple(5.47209, 23.17528),
+        std::make_tuple(19.68845, 5.73803),
+        std::make_tuple(7.86124, 1.65366)
+    };
+    params[35] = {
+        std::make_tuple(0.97231, 103.94730),
+        std::make_tuple(5.27765, 24.33975),
+        std::make_tuple(19.48822, 6.14586),
+        std::make_tuple(9.26182, 1.78869)
+    };
+    params[36] = {
+        std::make_tuple(0.96735, 107.28121),
+        std::make_tuple(5.09646, 25.51581),
+        std::make_tuple(19.25332, 6.56380),
+        std::make_tuple(10.68288, 1.92256)
+    };
+
+    return params[atomic_num][shell_num];
+}
+
+// Minimal Basis Iterative Stockhplder (JCTC, 2016, p. 3894–3912, Verstraelen et al.)
+std::tuple<SharedMatrix, SharedMatrix, SharedMatrix, SharedMatrix>
+PopulationAnalysisCalc::compute_mbis_multipoles(bool print_output) {
+
+    if (print_output) outfile->Printf("  ==> Computing MBIS Charges <==\n\n");
+    timer_on("MBIS");
+    
+    // => Setup 1RDM on DFTGrid <= //
+
+    Options& options = Process::environment.options;
+    const int max_iter = options.get_int("MBIS_MAXITER");
+    const double conv = options.get_double("MBIS_D_CONVERGENCE");
+    const int debug = options.get_int("DEBUG");
+
+    // Sets up MBIS Grid using MBIS_RAIDAL_POINTS and MBIS_SPHERICAL_POINTS options
+    std::map<std::string, int> mbis_grid_options_int;
+
+    mbis_grid_options_int["DFT_RADIAL_POINTS"] = options.get_int("MBIS_RADIAL_POINTS");
+    mbis_grid_options_int["DFT_SPHERICAL_POINTS"] = options.get_int("MBIS_SPHERICAL_POINTS");
+
+    std::map<std::string, std::string> mbis_grid_options_str;
+
+    mbis_grid_options_str["DFT_PRUNING_SCHEME"] = "NONE";
```

Comment:
Does is not work well with pruning? ("ROBUST" or "TREUTLER") Did you test it?

How about `DFT_WEIGHTS_TOLERANCE`?

Reply:
@hokru You bring up a good point, it may be a good idea to create a set of grid parameters exclusively for the MBIS grid, as we need to keep the MBIS grid independent of the DFT grid in case we are running DFT, and then MBIS on the molecule. I have not done much testing with pruning schemes or the `DFT_WEIGHTS_TOLERANCE` option. My teammate Zach suggested that I set `DFT_PRUNING_SCHEME` to NONE. What kinds of tests do you recommend?

Reply:
`DFT_WEIGHTS_TOLERANCE=-1` will turns off the screening of small weights, the current default (1e-15) is conservative, but for a test I'd try turning it off.

If it is sensible to seek computational efficiency for MBIS charges, then I'd try at least the `ROBUST` scheme.

Reply:
@hokru I tested the `ROBUST` and `TREUTLER` schemes on a tetracene molecule (6 cores), there were no significant time savings, while the quality of the integration remained the same (same electron discrepancy of 12E-5). I also tested setting `DFT_WEIGHTS_TOLERANCE -1.0`, the time was slightly worse while the quality of the integration did not change. (The MBIS charges were all within 0.001 electrons for each atom in all of the tests)

Timings:

Pruning schemes:
Treutler: ~50 seconds
Robust: ~52 seconds
None: ~52 seconds

DFT_WEIGHTS_TOLERANCE -1.0: ~55-65 seconds

Do you suggest that I add an `MBIS_PRUNING_SCHEME` option to MBIS, or any other options to the MBIS grid that can be set on a DFT grid (like `DFT_NUCLEAR_SCHEME` and `DFT_BASIS_TOLERANCE` http://www.psicode.org/psi4manual/master/dft.html#grid-selection)

Reply:
Turns out, the pruning was not activated the last time I tested it

I tested pruning again on tetracene (6 cores), and here are my results:

| **Scheme** | **Time (s)** | **Accurate Atomic Charges?** | **Electron Difference** |
| ------------ | ------------ | ------------------------------ | ----------------------- |
|     Robust      |     34.8         |                  Yes                             |              12E-5                 |
|     None         |     49.7         |                  Yes                             |              12E-5                 |
|     Treutler     |     27.3          |                  Yes                             |               2E-5                 |
|     None*       |     54.6          |                  Yes                             |               12E-5                |

Last Row = `DFT_WEIGHTS_TOLERANCE -1.0`


Seeing the "improvement" in both electron difference and time on the treutler scheme, I decided to test it on other molecules

Molecule, Difference (Treutler), Difference (None)
Benzene, 4E-5, 2E-5
Naphthalene, 4E-5, 7E-5
Anthracene, 3E-5, 9E-5
Tetracene, 2E-5, 12E-5
Pentacene, 1E-5, 14E-5

Therefore, I added `MBIS_PRUNING_SCHEME` as an option and set the default to `NONE`, should I instead set the default to `ROBUST` due to no difference in error and the relative safety of the robust scheme?

Reply:
Oh I didn't notice this. I would say "yes" to `ROBUST`. Not sure how others think about this.